pypika/queries.py

```diff
@@ -1172,15 +1172,17 @@ def replace_table(self, current_table, new_table):
 
 
 class JoinOn(Join):
-    def __init__(self, item, how, criteria):
+    def __init__(self, item, how, criteria, collate=None):
         super(JoinOn, self).__init__(item, how)
         self.criterion = criteria
+        self.collate = " COLLATE {}".format(collate) if collate else ""
```

Comment:
Would be better to produce the sql output in the `get_sql` function and store the raw values here in case you need access to how the query was produced.

Reply:
Yes you are right. I updated it.